fieldtracing/fieldtracing.cpp

```diff
@@ -763,392 +760,191 @@ namespace FieldTracing {
+    * enum TracingPointConnectionType {
+    *    CLOSED_CLOSED,
+    *    CLOSED_OPEN,
+    *    OPEN_CLOSED,
+    *    OPEN_OPEN,
+    *    CLOSED_DANGLING,
+    *    DANGLING_CLOSED,
+    *    OPEN_DANGLING,
+    *    DANGLING_OPEN,
+    *    DANGLING_DANGLING,
+    *    INVALID
+    * };
+    *
+    * Secondly, we are interested in finding out whether the seed point/DCCRG cell centre coordinate (again, too many fsgrid cells
+    * for sanity) are near/in a flux rope. As this relies also on tracing forward and backward along the field, we piggyback on the
+    * full box connection algorithm explained above.
+    * We trace along the field up to fieldTracingParameters.fluxrope_max_curvature_radii_to_trace*cellCurvatureRadius[n] and if
+    * within that tracing distance we have't extended further than
+    * fieldTracingParameters.fluxrope_max_curvature_radii_extent*cellCurvatureRadius[n] we are rolled up tightly enough to consider
+    * being close to a flux rope. We'll store the max extension reached into cellMaxExtension[n] for later fine-grained analysis.
+    * The arithmetic idea to avoid using yet more arrays:
+    * We use the cellFWConnection[n]/cellBWConnection[n]/cellConnection[n] (that can only be UNPROCESSED, CLOSED, OPEN or DANGLING,
+    * see TracingLineEndType enum above, ending with N_TYPES). As long as no full box tracing termination condition was reached
+    * we are at UNPROCESSED. If we exceed fieldTracingParameters.fluxrope_max_curvature_radii_to_trace*cellCurvatureRadius[n] OR
+    * fieldTracingParameters.fluxrope_max_m_to_trace we definitely are not near a flux rope and we mark this by adding N_TYPES to
+    * cellConnection[n].
+    * If we reach fieldTracingParameters.fluxrope_max_curvature_radii_to_trace*cellCurvatureRadius[n] without hitting the other
+    * thresholds or the inner/outer domain limits we are near/at a flux rope and we mark this by adding 2*N_TYPES to
+    * cellConnection[n].
+    * Later in the tracing when we reach a full box tracing termination condition we add CLOSED, OPEN or DANGLING to
+    * cellConnection[n]. If we reached such termination condition before the flux rope method reached a conclusion it's fine too,
+    * nothing has been added to cellConnection[n].
+    * At the very end, we check whether both cellFWConnection[n] and cellBWConnection[n] >= 2*TracingLineEndType::N_TYPES. If yes,
+    * this cell is near a fluxrope. This means we store the value of cellMaxExtension[n] into CellParams::FLUXROPE. Otherwise we
+    * store zero.
+    * After that we apply % TracingLineEndType::N_TYPES to recover values UNPROCESSED, CLOSED, OPEN, DANGLING in
+    * cellFWConnection[n] and cellBWConnection[n] so that we can assign the connection types for the full box connection described
+    * above.
+    *
+    * As a freebie since we computed the curvature anyway for flux rope tracing we store that into CellParams::CURVATUREX/Y/Z.
+    *
+    * \sa stepCellAcrossTaskDomain
     */
```

Comment:
How do you like this explanation attempt?

Reply:
Mostly understandable but I do have 2 comments which I left above and got a bit confused there